sdk/src/store.rs

```diff
@@ -1734,69 +1761,67 @@ impl Store {
         asset_path: &'a Path,
         validation_log: &mut impl StatusTracker,
     ) -> Result<()> {
-        let ext = get_supported_file_extension(asset_path).ok_or(Error::UnsupportedType)?;
-
-        let cai_loader = get_cailoader_handler(&ext).ok_or(Error::UnsupportedType)?;
-
-        let mut asset_reader = fs::File::open(asset_path)?;
-
-        // read xmp if available
-        let xmp_opt = cai_loader.read_xmp(&mut asset_reader);
-
-        let xmp_copy = xmp_opt.clone();
-
-        Store::verify_store(
-            self,
-            xmp_opt,
-            &ClaimAssetData::PathData(asset_path),
-            validation_log,
-        )?;
-
-        // set the provenance if there is xmp otherwise it will default to active manifest
-        if let Some(xmp) = xmp_copy {
-            if let Some(xmp_provenance) = extract_provenance(&xmp) {
-                let claim_label = Store::manifest_label_from_path(&xmp_provenance);
-                self.set_provenance_path(&claim_label);
-            }
-        }
-
-        Ok(())
+        Store::verify_store(self, &ClaimAssetData::PathData(asset_path), validation_log)
     }
 
     // verify from a buffer without file i/o
     pub fn verify_from_buffer(
         &mut self,
         buf: &[u8],
-        asset_type: &str,
+        _asset_type: &str,
         validation_log: &mut impl StatusTracker,
     ) -> Result<()> {
-        let mut buf_reader = Cursor::new(buf);
+        Store::verify_store(self, &ClaimAssetData::ByteData(buf), validation_log)
+    }
 
-        let cai_loader = get_cailoader_handler(asset_type).ok_or(Error::UnsupportedType)?;
+    // fetch remote manifest if possible
+    #[cfg(not(target_arch = "wasm32"))]
+    fn fetch_remote_manifest(url: &str) -> Result<Vec<u8>> {
+        use conv::ValueFrom;
+        use std::io::Read;
+
+        //const MANIFEST_CONTENT_TYPE: &str = "application/x-c2pa-manifest-store"; // todo verify once these are served
+        const DEFAULT_MANIFEST_RESPONSE_SIZE: usize = 10 * 1024 * 1024; // 10 MB
```

Comment:
I don't know what a reasonable limit would be for this, but I guess we will find out.

Reply:
I am going to expose this as an option when I add the config stuff.